### PR TITLE
Implement order_by

### PR DESCRIPTION
**Pull request by marcofucci**
- Klass.objects.all().order_by('...') makes a cdms call ordered by the cdms mapped field
- support for single, multiple field, asc and desc
- ordering by a non-cdms field is not allowed
- Klass.objects.skip_cdms().order_by('...') skips cdms calls

It's important that the Django model has a default ordering otherwise underlying cdms calls will not be ordered the same way as local ones.
A warning will be given if no default ordering is specified in Meta.

### CODE REVIEW

**Comment by marcofucci** [comment]
@danielquinn did you get the email?

File: crm-poc/apps/migrator/query.py [hunk 1]
```diff
@@ -76,8 +76,46 @@ def execute(self):
...
+    def get_order_by(self):
+        cdms_orderby = []
+
+        if self.query.order_by:
+            ordering = self.query.order_by
+        else:
+            ordering = self.query.model._meta.ordering
+            if not ordering:
+                warnings.warn(
+                    "{0} does not have a default ordering so cdms calls and local ones "
```
Comment (danielquinn) [inline crm-poc/apps/migrator/query.py]: You don't need the `0` here, as that's only necessary for Python 2.6.  2.7+ allows you to just use `{}`.

File: crm-poc/apps/migrator/query.py [hunk 2]
```diff
@@ -76,8 +76,46 @@ def execute(self):
...
+            col, order = get_order_dir(field, 'ASC')
+            try:
+                cdms_field = self.get_migrator().get_cdms_field(col)
+            except NotMappingFieldException:
+                raise NotImplementedError(
+                    'Cannot order by {0}, only ordering by direct fields currently implemented'.format(col)
+                )
+
+            cdms_orderby.append(
+                '{0} {1}'.format(cdms_field.cdms_name, order.lower())
```
Comment (danielquinn) [inline crm-poc/apps/migrator/query.py]: Same here: `'{} {}'.format(cdms_field.cdms_name, order.lower())` would work too.

**Comment by danielquinn** [comment]
Everything looks good.  I can merge now unless you wanna change those nit-picky comments I made about `{}`.

**Comment by marcofucci** [comment]
Do you really like `{}` ? can change if you want but kinda prefer {n}, it's nicer no?

**Comment by danielquinn** [comment]
I have no strong feelings about it :-)

**Pull request merged**